indico_zodbimport/modules/event_timetable.py

```diff
@@ -194,10 +197,79 @@ def _process_references(self, reference_cls, old_object):
+        email = getattr(old_person, '_email', None)
+        email = sanitize_email(convert_to_unicode(email).lower()) if email else email
+        if not is_valid_mail(email, False):
+            email = None
+            if not skip_empty_email:
+                self.importer.print_warning(cformat('%{yellow!}Skipping invalid email {}').format(email),
+                                            event_id=self.event.id)
+        if not email and skip_empty_email:
+            return None
+        person = EventPerson(event_new=self.event if with_event else None,
+                             email=email, **self._get_person_data(old_person))
+        if not person.first_name and not person.last_name:
+            self.importer.print_warning(cformat('%{yellow!}Skipping nameless event person'), event_id=self.event.id)
+            return None
+        return person
+
+    def _get_person(self, old_person):
+        email = getattr(old_person, '_email', None)
+        email = sanitize_email(convert_to_unicode(email).lower()) if email else email
+        return self.event_person_map[email] if email else self._create_person(old_person, with_event=True)
+
+    def _get_person_data(self, old_person):
+        data = {}
+        for old_attr, new_attr in PERSON_INFO_MAP.iteritems():
+            data[new_attr] = convert_to_unicode(getattr(old_person, old_attr, ''))
+        data['_title'] = USER_TITLE_MAP.get(getattr(old_person, '_title', ''), UserTitle.none)
+        return data
+
     def _migrate_event_persons(self):
+        all_persons = defaultdict(list)
+        old_people = []
+        for chairperson in getattr(self.old_event, '_chairs', []):
+            old_people.append(chairperson)
+        for old_contrib in self.old_event.contributions.itervalues():
+            for speaker in getattr(old_contrib, '_speakers', []):
+                old_people.append(speaker)
+            for author in getattr(old_contrib, '_primaryAuthors', []):
+                old_people.append(author)
+            for coauthor in getattr(old_contrib, '_coAuthors', []):
+                old_people.append(coauthor)
+            for old_subcontrib in old_contrib._subConts:
+                for speaker in getattr(old_subcontrib, 'speakers', []):
+                    old_people.append(speaker)
+        for old_entry in self.old_event._Conference__schedule._entries:
+            entry_type = old_entry.__class__.__name__
+            if entry_type == 'LinkedTimeSchEntry':
+                old_block = old_entry._LinkedTimeSchEntry__owner
+                for convener in getattr(old_block, '_conveners', []):
+                    old_people.append(convener)
+        for old_person in old_people:
+            person = self._create_person(old_person, skip_empty_email=True)
+            if person:
+                all_persons[person.email].append(person)
+        for email, persons in all_persons.iteritems():
+            person = EventPerson(email=email,
```

Comment:
yeah, makes sense - forgot that we'll still override stuff in the links to end up with the correct data